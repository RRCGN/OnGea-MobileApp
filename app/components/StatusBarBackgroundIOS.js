/**
 * Background for the translucent StatusBar in iOS as a Linear Gradient.
 * Returns no Component in Android.
 * @flow
 */

import React from 'react'
import { StyleSheet, Platform } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'


type Props = {
  type?: 'light' | 'dark'
}

const StatusBarBackgroundIOS = ({ type = 'light' }: Props) => {
  if (Platform.OS === 'ios') {
    return (
      <LinearGradient
        colors={['rgba(0, 0, 0, 0.3)', 'rgba(0, 0, 0, 0)']}
        style={styles.background}
      />
    )
  }

  return null
}

export default StatusBarBackgroundIOS


const styles = StyleSheet.create({
  background: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    height: 60,
    zIndex: 1,
  }
})
