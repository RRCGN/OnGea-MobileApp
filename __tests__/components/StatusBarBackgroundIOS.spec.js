import 'react-native'
import React from 'react'
import StatusBarBackgroundIOS from '../../src/components/StatusBarBackgroundIOS'

import renderer from 'react-test-renderer'

describe('<StatusBarBackgroundIOS />', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <StatusBarBackgroundIOS />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
