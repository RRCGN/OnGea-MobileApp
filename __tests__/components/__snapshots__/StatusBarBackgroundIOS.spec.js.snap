exports[`<StatusBarBackgroundIOS /> renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      1291845632,
      0,
    ]
  }
  end={undefined}
  locations={null}
  start={undefined}
  style={
    Object {
      "height": 60,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1,
    }
  } />
`;
